### ISSUE TITLE
Invalid cast. The supplied value [object Object] did not pass the test 'isParameter'.

### ISSUE THREAD

**Issue by cevek** [Microsoft/TypeScript#25873]
**TypeScript Version:**  3.1.0-dev.20180717

**Search Terms:**

**Code**

```ts
Error: Debug Failure. Invalid cast. The supplied value [object Object] did not pass the test 'isParameter'.
    at Object.cast (/usr/local/lib/node_modules/typescript/lib/tsserver.js:1291:22)
    at doChange (/usr/local/lib/node_modules/typescript/lib/tsserver.js:106695:40)
    at /usr/local/lib/node_modules/typescript/lib/tsserver.js:106634:109
    at Function.ChangeTracker.with (/usr/local/lib/node_modules/typescript/lib/tsserver.js:102904:17)
    at Object.getCodeActions (/usr/local/lib/node_modules/typescript/lib/tsserver.js:106634:64)
    at /usr/local/lib/node_modules/typescript/lib/tsserver.js:103859:121
    at Object.flatMap (/usr/local/lib/node_modules/typescript/lib/tsserver.js:498:25)
    at Object.getFixes (/usr/local/lib/node_modules/typescript/lib/tsserver.js:103859:23)
    at /usr/local/lib/node_modules/typescript/lib/tsserver.js:111706:35
    at Object.flatMap (/usr/local/lib/node_modules/typescript/lib/tsserver.js:498:25)
    at Object.getCodeFixesAtPosition (/usr/local/lib/node_modules/typescript/lib/tsserver.js:111704:23)
    at IOSession.Session.getCodeFixes (/usr/local/lib/node_modules/typescript/lib/tsserver.js:119278:64)
    at Session.handlers.ts.createMapFromTemplate._a.(anonymous function) (/usr/local/lib/node_modules/typescript/lib/tsserver.js:118152:61)
    at /usr/local/lib/node_modules/typescript/lib/tsserver.js:119442:88
    at IOSession.Session.executeWithRequestId (/usr/local/lib/node_modules/typescript/lib/tsserver.js:119433:28)
    at IOSession.Session.executeCommand (/usr/local/lib/node_modules/typescript/lib/tsserver.js:119442:33)
    at IOSession.Session.onMessage (/usr/local/lib/node_modules/typescript/lib/tsserver.js:119462:35)
    at Interface.<anonymous> (/usr/local/lib/node_modules/typescript/lib/tsserver.js:120735:27)
```

**Expected behavior:**

**Actual behavior:**

**Playground Link:** 

**Related Issues:**

**Comment by andy-ms** [comment]
Ought to have been fixed along with Microsoft/vscode#45785. That diagnostic is only ever added at a ParameterDeclaration.

**Comment by sheetalkamat** [comment]
@cevek Can you try now and see if this is fixed. if not please provide repro and more details by sharing tsserver log. Thanks.
